### ISSUE TITLE
Implementing an Event Listener for Detecting End of Typing in a React Typist Component

### ISSUE THREAD

**Issue by magedhelmy1** [BraisC/react-text-typist#13]
I need a way to detect when Typer has finished typing.

In my app, I would like to disable the "Request" button until Typist finishes typing to prevent spam clicking. How can I do that?

Couple of ways I thought I might be implemented (but since its not supported in the package yet) are as follows

```
              <Typer
                sentences={[output]}
                startDelay={0}
                typingSpeed={7}
                pauseTime={250}
                cursorColor="black"
                showCursor={true}
                hideCursorOnFinish={true}
                loop={false}
                onEvent={(event) => {
                  if (event === "startTyping") {
                    setTyping(true);
                  } else if (event === "finishTyping") {
                    setTyping(false);
                  }
                }}
              />
```

or

```
import Typer, { TyperEvent } from "react-text-typist";

              <Typer
                sentences={[output]}
                startDelay={0}
                typingSpeed={7}
                pauseTime={250}
                cursorColor="black"
                showCursor={true}
                hideCursorOnFinish={true}
                loop={false}
                onTyped={() => setTyping(true)}
                onFinishedTyping={() => setTyping(false)}
              />
```

Please let me know if there is a way to do this already.

Thank you

**Comment by MarshallChang** [comment]
#14

**Comment by BraisC** [comment]
Hey, I am going to take a look at the PR by @MarshallChang ASAP and come up with a solution for this.
Meanwhile, as you know the typing speed, the pause time and the number of characters of your string, you can calculate how long will it take to type it and do a setTimeout:

```javascript
const timeToWrite = numberOfChars * typingSpeed + pauseTime;

setTimeout(() => {
  setTyping(false)
}, timeToWrite);
```

I know, it is not pretty, but can work in the meantime if you are in a hurry

**Comment by skyliwq** [comment]
++++++++

**Comment by skyliwq** [comment]
I really need this feature

**Comment by skyliwq** [comment]
I really need this feature

**Comment by BraisC** [comment]
Sorry for the delay, it is now implemented in v1.1.8, the property is called `onTypeFinish` and it receives a function

Enjoy

**Comment by magedhelmy1** [comment]
Thank you!
